ddl/cancel_test.go

```diff
@@ -73,16 +73,33 @@ var allTestCase = []testCancelJob{
 	{"create table test_create_table(a int)", true, model.StateNone, true, false, nil},
 	{"create table test_create_table(a int)", false, model.StatePublic, false, true, nil},
+	// Drop table.
+	{"drop table test_create_table", true, model.StateNone, true, false, nil},
+	{"drop table test_create_table", false, model.StateWriteOnly, true, true, []string{"create table if not exists test_create_table(a int)"}},
+	{"drop table test_create_table", false, model.StateDeleteOnly, true, true, []string{"create table if not exists test_create_table(a int)"}},
+	{"drop table test_create_table", false, model.StatePublic, false, true, []string{"create table if not exists test_create_table(a int)"}},
 	// Create schema.
 	{"create database test_create_db", true, model.StateNone, true, false, nil},
 	{"create database test_create_db", false, model.StatePublic, false, true, nil},
+	// Drop schema.
+	{"drop database test_create_db", true, model.StateNone, true, false, nil},
+	{"drop database test_create_db", false, model.StateWriteOnly, true, true, []string{"create database if not exists test_create_db"}},
+	{"drop database test_create_db", false, model.StateDeleteOnly, true, true, []string{"create database if not exists test_create_db"}},
+	{"drop database test_create_db", false, model.StatePublic, false, true, []string{"create database if not exists test_create_db"}},
 	// Drop column.
 	{"alter table t drop column c3", true, model.StateNone, true, false, nil},
 	{"alter table t drop column c3", false, model.StateDeleteOnly, true, false, nil},
 	{"alter table t drop column c3", false, model.StateDeleteOnly, false, true, []string{"alter table t add column c3 bigint"}},
 	{"alter table t drop column c3", false, model.StateWriteOnly, true, true, []string{"alter table t add column c3 bigint"}},
 	{"alter table t drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t add column c3 bigint"}},
 	{"alter table t drop column c3", false, model.StatePublic, false, true, []string{"alter table t add column c3 bigint"}},
+	// Drop column with index.
+	{"alter table t drop column c3", true, model.StateNone, true, false, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateDeleteOnly, true, false, nil},
+	{"alter table t drop column c3", false, model.StateDeleteOnly, false, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateWriteOnly, true, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StatePublic, false, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
```

Comment:
Why can't this DDL be canceled? The state is public.

I have the same question for line 95.

Reply:
Added TODO to fix them.